notebooks/Threshold Optimization Postprocessing for Binary Classification.ipynb

```diff
@@ -0,0 +1,452 @@
+    "fairness_aware_predictions_EO_train = postprocessed_predictor_EO.predict(X_train, sensitive_features=sensitive_features_train)\n",
+    "fairness_aware_predictions_EO_test = postprocessed_predictor_EO.predict(X_test, sensitive_features=sensitive_features_test)\n",
+    "\n",
+    "show_proportions(X_train, sensitive_features_train, fairness_aware_predictions_EO_train, y_train, description=\"equalized odds with postprocessed model on training data:\", plot_row_index=1)\n",
+    "show_proportions(X_test, sensitive_features_test, fairness_aware_predictions_EO_test, y_test, description=\"equalized odds with postprocessed model on test data:\", plot_row_index=2)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predicted_ys = [\n",
+    "    unconstrained_predictor.predict(X_test),\n",
+    "    postprocessed_predictor_EO.predict(X_test, sensitive_features=sensitive_features_test)\n",
+    "]\n",
+    "FairlearnDashboard(sensitive_features=sensitive_features_test.values,\n",
+    "                   true_y=y_test.values,\n",
+    "                   predicted_ys=unconstrained_predictor.predict(X_test),\n",
+    "                   class_names=None,\n",
+    "                   feature_names=X_test.columns.values.tolist(),\n",
+    "                   is_classifier=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Post Processing in Detail\n",
+    "\n",
+    "While this worked as the numbers show, it's not entirely obvious how it found its solution. The following section will provide a deep dive on post processing for Equalized Odds (EO).\n",
+    "\n",
+    "Our post processing method (based on work by [Hardt, Price, Srebro](https://arxiv.org/pdf/1610.02413.pdf)) takes a fairness-unaware model and disparity constraints (such as EO) in the constructor and features (X), labels (y), and a sensitive feature (sensitive_features) in the fit method. It subsequently uses the model to make predictions for all samples in X. Note that these predictions could be 0/1 (as in our example), or more categories, or even real valued scores.\n",
+    "In our case, this looks as follows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "scores = unconstrained_predictor_wrapper.predict(X_train)\n",
+    "scores"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Finding threshold rules\n",
+    "\n",
+    "The algorithm then tries to find all thresholding rules with which it could divide the data. Any score for which the thresholding rule evaluates to true is predicted to be 1. It does that for each group, so in this case for each race. Depending on your scores you could have lots of thresholding rules, between each set of such values. For each rule we just evaluate the following two probabilities empirically:\n",
```

Comment:
I'd try to avoid 'your' and the 2nd person, keep it more objective-y with 'the dataset's' or something

Reply:
replace all "your"s and "our"s. Great recommendation!